### PR TITLE
Support for getting response headers (fix issue #23)

### PR DESCRIPTION
**Pull request by tcfialho**
Implementing support for getting response headers as requested in issue #23 most voted suggestion.

To get the response metadata you should wrap your return type with RefitResponse<T>

Task<RefitResponse<User>> GetUserWithMetadata(string userName);
or
IObservable<RefitResponse<User>> GetUserObservableWithMetadata(string userName);

### CODE REVIEW

File: Refit/VV-Refit.csproj [hunk 1]
```diff
@@ -2,7 +2,9 @@
```
Comment (Cheesebaron) [inline Refit/VV-Refit.csproj]: This project should probably not be renamed.

File: Refit/VV-Refit.csproj [hunk 2]
```diff
@@ -2,7 +2,9 @@
 
   <PropertyGroup>
     <Product>Refit ($(TargetFramework))</Product>
-    <TargetFrameworks>netstandard1.4;net45</TargetFrameworks>
+    <TargetFramework>netcoreapp2.0</TargetFramework>
```
Comment (Cheesebaron) [inline Refit/VV-Refit.csproj]: TargetFramework should not be changed

**Comment by onovotny** [comment]
Hi,

I'm happy to take this PR, but why is there a Jenkins file added? We don't use Jenkins and have no plans to maintain that.

Can you please remove that from this PR?

File: Refit/VV-Refit.csproj [hunk 3]
```diff
@@ -2,7 +2,9 @@
...
+	  <PackageVersion>1.0.0</PackageVersion>
```
Comment (Cheesebaron) [inline Refit/VV-Refit.csproj]: No need to add package version as it is patched by the build

File: Jenkinsfile [hunk 4]
```diff
@@ -0,0 +1,34 @@
+node('Slave-Azure'){
```
Comment (Cheesebaron) [inline Jenkinsfile]: Not sure you want to add this jenkins file with credentials...

**Comment by bennor** [comment]
I'm not sure why this was closed, but it feels to me that it's not that much of an improvement on just returning `Task<HttpResponseMessage>` and deserializing the content yourself. It feels like it adds a lot of work to manually copy most of the HttpResponse. 

I'd personally rather see support for tuples so you could do:

```
[Get("/api/whatever")]
Task<(HttpStatusCode, WhateverDto)>GetWhatever();
```

You could also support getting headers by including a value of type `HttpResponseHeaders` and/or `HttpContentHeaders` in your tuple.

**Comment by onovotny** [comment]
@bennor I like the idea of using Tuples for this, but do you think it's extra pollution to be that granular? What if it's just `HttpResponseMessage` and `T` as possible options? Once you have the `HttpReponseMessage`, you can get the status code, headers or anything else out of it.

File: Refit/VV-Refit.csproj (hunk 3 above)
Comment (tcfialho) [reply Refit/VV-Refit.csproj]: Please remove that i made another PR. #414

File: Refit/VV-Refit.csproj (hunk 2 above)
Comment (tcfialho) [reply Refit/VV-Refit.csproj]: Please ignore that pull request. I made another with all fixed. #414

File: Jenkinsfile (hunk 4 above)
Comment (tcfialho) [reply Jenkinsfile]: I made another pull request with all fixed. #414

File: Refit/VV-Refit.csproj (hunk 1 above)
Comment (tcfialho) [reply Refit/VV-Refit.csproj]: Due some mistakes, I made another pull request with all fixed. #414

File: Refit/VV-Refit.csproj (hunk 2 above)
Comment (tcfialho) [reply Refit/VV-Refit.csproj]: Due some mistakes, I made another pull request with all fixed. #414

File: Refit/VV-Refit.csproj (hunk 3 above)
Comment (tcfialho) [reply Refit/VV-Refit.csproj]: Due some mistakes, I made another pull request with all fixed. #414

File: Jenkinsfile (hunk 4 above)
Comment (tcfialho) [reply Jenkinsfile]: Due some mistakes, I made another pull request with all fixed. #414

**Comment by tcfialho** [comment]
Due some mistakes, I made another pull request with all fixed. #414

**Comment by bennor** [comment]
@onovotny Yeah, you're probably right.

**Comment by onovotny** [comment]
@bennor Actually, I take that back. The `RefitResponse<T>` may be the best solution here since `HttpResponseMessage` contains the internal content stream and that may not be able to be read twice. Encapsulating the response message prevents the double-read.

**Comment by bennor** [comment]
👍 

I've started a review on the other PR anyway.

**Pull request closed without merging**